source/common/quic/envoy_quic_utils.cc

```diff
@@ -143,8 +143,14 @@ createConnectionSocket(const Network::Address::InstanceConstSharedPtr& peer_addr
   if (local_addr == nullptr) {
     local_addr = Network::Utility::getLocalAddress(peer_addr->ip()->version());
   }
+  size_t max_addresses_cache_size =
+      Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.quic_upstream_socket_use_address_cache_for_read")
+          ? 4u
```

Comment:
4 seems low. This should probably have a config knob to tune the size (which I'd be fine adding in a followup PR if you don't want to increase the scope of this one). How did you decide on this value for the default?

Reply:
This is just for client connection which only has self/peer addresses for the current path and at most an alternative path. QUICHE impl only stores upto 2 pathes.

Reply:
Got it; I missed that this was only in the client path. 4 makes sense then in this context.